src/peft/tuners/oft/model.py

```diff
@@ -76,33 +90,283 @@ class OFTModel(LycorisTuner):
 
     def _create_and_replace(
         self,
-        config: LycorisConfig,
-        adapter_name: str,
-        target: Union[OFTLayer, nn.Module],
-        target_name: str,
-        parent: nn.Module,
-        current_key: str,
-    ) -> None:
+        oft_config,
+        adapter_name,
+        target,
+        target_name,
+        parent,
+        current_key,
+        **optional_kwargs,
+    ):
+        if current_key is None:
+            raise ValueError("Current Key shouldn't be `None`")
+
+        bias = hasattr(target, "bias") and target.bias is not None
+        kwargs = {
+            "r": oft_config.r,
+            "oft_block_size": oft_config.oft_block_size,
+            "module_dropout": oft_config.module_dropout,
+            "coft": oft_config.coft,
+            "eps": oft_config.eps,
+            "block_share": oft_config.block_share,
+            "fan_in_fan_out": oft_config.fan_in_fan_out,
+            "init_weights": oft_config.init_weights,
+        }
+        kwargs["bias"] = bias
+
+        # If it is not a OFTLayer, create a new module, else update it with new adapters
+        if not isinstance(target, OFTLayer):
+            new_module = self._create_new_module(oft_config, adapter_name, target, **kwargs)
+            if adapter_name not in self.active_adapters:
+                # adding an additional adapter: it is not automatically trainable
+                new_module.requires_grad_(False)
+            self._replace_module(parent, target_name, new_module, target)
+        else:
+            target.update_layer(
+                adapter_name,
+                r=oft_config.r,
+                oft_block_size=oft_config.oft_block_size,
+                module_dropout=oft_config.module_dropout,
+                coft=oft_config.coft,
+                eps=oft_config.eps,
+                block_share=oft_config.block_share,
+                init_weights=oft_config.init_weights,
+            )
+
+    def _replace_module(self, parent, child_name, new_module, child):
+        setattr(parent, child_name, new_module)
+        # It's not necessary to set requires_grad here, as that is handled by
+        # _mark_only_adapters_as_trainable
+
+        # child layer wraps the original module, unpack it
+        if hasattr(child, "base_layer"):
+            child = child.base_layer
+
+        if not hasattr(new_module, "base_layer"):
+            new_module.weight = child.weight
+            if hasattr(child, "bias"):
+                new_module.bias = child.bias
+
+        if getattr(child, "state", None) is not None:
+            if hasattr(new_module, "base_layer"):
+                new_module.base_layer.state = child.state
+            else:
+                new_module.state = child.state
+            new_module.to(child.weight.device)
+
+        # dispatch to correct device
+        for name, module in new_module.named_modules():
+            if self.prefix in name:
+                module.to(child.weight.device)
```

Comment:
```suggestion
        meta = torch.device("meta")
        # dispatch to correct device
        for name, module in new_module.named_modules():
            if self.prefix in name:
                if not any(p.device == meta for p in module.parameters()):
                    module.to(child.weight.device)
```

This will fix the `low_cpu_mem_usage` errors.